File: backend/src/main/java/bag/service/auth/AuthService.java
package bag.service.auth;

import bag.modal.dto.AuthResponse;
import bag.modal.request.AuthRequest;
import bag.modal.request.GoogleAuthRequest;

public interface AuthService {
    AuthResponse login(AuthRequest request);
    AuthResponse googleLogin(GoogleAuthRequest request);
    AuthResponse refreshToken(String refreshToken);
    void logout(int accountId);
    String resentOtp(String email, String action);
}

File: frontend/bagshop/src/service/AuthService.js
import axios from "axios";

const axiosRaw = axios.create({
  baseURL: import.meta.env.VITE_API_BASE_URL || "http://localhost:8081/",
  withCredentials: true,
  headers: { "Content-Type": "application/json" },
});

const AuthService = {
  login(loginRequest) {
    return axiosRaw.post("auth/login", loginRequest);
  },
  logout(accountId) {
    return axiosRaw.post(`auth/logout/${accountId}`);
  },
  refresh() {
    return axiosRaw.post("auth/refresh");
  },
  resentOtp(email, action) {
    return axiosRaw.post(`auth/resent?email=${email}&action=${action}`);
  },
  googleLogin(accessToken) {
    return axiosRaw.post("auth/google", { accessToken });
  },
};

export default AuthService;
